woocommerce/woocommerce-android
# Allow private products in order creation

Summary
==========
Fix issue #9509 by updating the Order Creation product restrictions to allow private products.

Private Products are a visibility option in the Product form in wp-admin, whenever you set the visibility to Private, the Status of the product will also change to `Privately Published` automatically.

<img width="284" alt="image" src="https://github.com/user-attachments/assets/6a11732f-2beb-4ded-a22d-df0743b8c4b7">

How
==========
WCAndroid uses a `ProductRestrictions` interface to create Product to compose filters that can select which products we want to display in a give section of the app. This PR introduces a `NonPurchasableProducts` restriction and replaces the `NonPublishedProducts` filter in the Order Creation scenario.

The decision to not update the `NonPublishedProducts` to allow Private product is based on the fact that multiple areas of the app uses this Restriction definition, and updating it would cause side-effects.

Screen capture
==========
https://github.com/user-attachments/assets/93f29e79-825d-413a-98f9-548c08c674a0

How to Test
==========
1. Create a 'Private' product
2. Look in the app to verify the product is shown in the 'Products' tab
3. Go to 'Orders' to create a new order
4. Tap "+ Add products" and search for that product

Update release notes:
